### PR TITLE
Sighash_token / allowing implicit burns

### PR DESCRIPTION
**Pull request by imaginaryusername**
Change specification to allow implicit burns. Importantly, introduce a new sighash flag to prevent unaware wallets from accidentally burning token outputs sent to them. Maintains expectations for wallet safety.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -445,13 +468,13 @@ Mutable tokens allow the holder to create **only one new token** (i.e. "modify"
...
+### Safety from Accidental Destruction while Allowing Implicit Burns
 
-### Disallowing Implicit Destruction of Immutable Tokens
+This specification allows spent tokens from being implicitly destroyed by not re-including them in a transaction's outputs. Similar to BCH transactions, omission of any amount from the spent outputs in created UTXOs is allowed; unlike BCH transactions, such omissions are not claimable by miners as mining fees. This is a modest efficiency gain for most tokens as workarounds like OP_RETURN or burn addresses are not required. For mutable tokens where covenant usecases are especially important, not requiring old tokens to be deposited to burn outputs every time they're mutated presents a larger gain.
```
Comment (emergent-reasons) [inline readme.md:473]:
```suggestion
This specification allows spent tokens to be implicitly destroyed by not re-including them in a transaction's outputs. Similar to BCH transactions, omission of any amount from the spent outputs in created UTXOs is allowed; unlike BCH transactions, such omissions are not claimable by miners as mining fees. This is a modest efficiency gain for most tokens as workarounds like OP_RETURN or burn addresses are not required. For mutable tokens where covenant usecases are especially important, not requiring old tokens to be deposited to burn outputs every time they're mutated presents a larger gain.
```

Comment (imaginaryusername) [reply readme.md]: thanks, fixed in https://github.com/bitjson/cashtokens/pull/14/commits/7258d69491696daecf4896528c87ee908bb6605e

**Comment by imaginaryusername** [comment]
Note: Upon implementation there seems to be additional complexity involved if we put the flag on Sighash bits - will need to investigate further

File: readme.md [hunk 2]
```diff
@@ -160,6 +160,29 @@ The following test vectors demonstrate valid and invalid `PREFIX_TOKEN` encoding
 
 </details>
 
+### SIGHASH_TOKEN
+
+Any output that carries `PREFIX_TOKEN` MUST only be spent if its spending input is signed with bit 5 set on its nHashType - this is known as `SIGHASH_TOKEN`, 0x20. Outputs that do not carry `PREFIX_TOKEN` MUST NOT be spent if signed with the `SIGHASH_TOKEN` bit set.
```
Comment (cculianu) [inline readme.md:165]: >  Outputs that do not carry `PREFIX_TOKEN` MUST NOT be spent if signed with the `SIGHASH_TOKEN` bit set.

Wait -- so it's either/or? This means you cannot mix a non-token and a token spend in the same tx? Are you sure we want this?!  Am I reading this right?!?! Wouldn't this make it very hard or maybe impossible to spend tokens if they are say at the dust limit of BCH?!  You can never fund the txn to pay fees!!

**Comment by bitjson** [comment]
Closed by https://github.com/bitjson/cashtokens/issues/11

**Comment by bitjson** [comment]
Thank you again for all your work on this @imaginaryusername!

**Pull request closed without merging**